server/src/main/java/org/jboss/as/server/parsing/SystemPropertiesXml.java

```diff
@@ -130,6 +133,18 @@ void parseSystemProperties(final XMLExtendedStreamReader reader, final ModelNode
             if (!setName) {
                 throw ParseUtils.missingRequired(reader, Collections.singleton(NAME));
             }
+
+            Properties properties = WildFlySecurityManager.getSystemPropertiesPrivileged();
```

Comment:
you could read the system properties outside the `while` block

Reply:
I have moved it outside.